### ISSUE TITLE
Motion Path - Center editing not working.

### ISSUE THREAD

**Issue by gab3d** [opentoonz/opentoonz#2708]
From the manual:
> To change the center of the object:
>    1. Choose the Animate tool (animate).
>    2. Move the center of the object to the new position.
>    3. Right-click the object node in the Stage Schematic and select Reset Center from the menu that opens.

...but that does not work at all.
With step 2, you move the Center off the path.
When you do step 3. You effectively reset the center, and it returns to be over the path, undoing what step 2 did.
🤔

**Comment by RodneyBaker** [comment]
It seems to me that the process is correct but the documentation is wrong.
We should expect the Center to reset upon activating 'Reset Center'.

Perhaps documentation should drop Step 3?

**Comment by gab3d** [comment]
mmm... I don't think so.
To me, this procedure should allow for moving the center, and then offset both (the center and the column) the right amount back, so that the center would be again on the motion path, but with the drawing offset... or something! 😄

**Comment by gab3d** [comment]
@shun-iwasawa  any insights on how this is supposed to work?

**Comment by manongjohn** [comment]
Looking at the logic regarding Reset Center, I believe this is just a badly worded statement in the document.   The Reset Center is supposed to put the center back to the original starting position as if you never moved it to begin with.

I think in the documentation there are 2 places that should be reworded 

>If you want to change the center of the object, you can move it using the Animate tool (animate).  **To reset the position back to normal**, use the Reset Center command in the Stage Schematic.

and

>3. **To reset the position back to normal,** right-click the object node in the Stage Schematic and select Reset Center from the menu that opens.

**Comment by gab3d** [comment]
@manongjohn 
thanks for investigating!

I know it sound as the most logical of things that the manual should be wrong, I know it is 100% correct. (I know it because I have Toonz and it does exactly what the manual describes). 🤔 

Here's a capture of the feature in action:
[mpath_center_toonz.zip](https://github.com/opentoonz/opentoonz/files/3648980/mpath_center_toonz.zip)

**Comment by RodneyBaker** [comment]
Thanks for the video @gab3d 
That helps a lot.

That looks to me like a feature of motion paths that must have been removed at some point (over on the Studio Ghibli side pior to release of Opentoonz).  Perhaps it can be reverted back.

**Comment by manongjohn** [comment]
Sorry. I don't have Toonz to compare against so I have no frame of reference of what it was supposed to do.

In Toonz, if the level was not connected to a Motion Path, what does Reset Center do?  Does it move the center point back the center and move the drawing with it?

To me Reset Center from a non-Motion Path point of view, moves the center back to where it started. Though I suppose it's debatable if that should also move the drawing relative to the center being moved, but that's just me.  

Perhaps that was the original behavior and it somehow broke along the way?  I wonder if it was working in older OT versions?

At any rate, this doesn't match the documentation, I'd consider this a bug.

**Comment by RodneyBaker** [comment]
I'm not seeing the Reset Center option for Motion Paths in v1.1.2 (circa Nov 2016),
I haven't tried earlier releases.

**Comment by gab3d** [comment]
@manongjohn 

well, that's what it actually does:

[reset_center_toonz.zip](https://github.com/opentoonz/opentoonz/files/3649411/reset_center_toonz.zip)

**Comment by manongjohn** [comment]
Then this is not Motion Path specific. It's just a Reset Center command issue/expectation.

In my opinion, without knowing Toonz, this command works as expected. The explanation of the command doesn't say it will recenter the drawing based on center point being relocated back to origin unless I missed it somewhere. I guess that can be implied but "Reset" means to start over so I wouldn't expect that behavior.

Guess the question is should we fix it to match Toonz or add a new command to Reset View on Center Point?

**Comment by gab3d** [comment]
@manongjohn 
I agree the "Reset Center" concept normally used everywhere is what we currently have now in OT; in that sense I guess we should preserve it, as it allows to perform a needed function and it's trivial to understand as it is.

Nevertheless, the behavior seen in Toonz is undoubtedly useful too, and if possible I'd love to see it implemented as a separate command. 
As for its name, I'm not sure... *Recenter on Center*? 😸 idk, what is really doing is compensating for the center offset introduced while moving the Center.

**Comment by RodneyBaker** [comment]
I've been thinking about this issue on and off for awhile.
One thought I had was of how we might parent the object in question to another object (such as a peg or column) and then have it fulfill the requirement.

I'm talking in the abstract here via workaround but also with an eye for implementing the fuller fix also.

A reason I am considering the parenting is also to deal with the issue of the Center point (or pivot) of an object not being animateable.  Where the parent of the object is treated as the pivot/offset that parent can be animated.

Apologies for the stray thoughts but I was working with motion paths and thought I'd drop again on this issue and read through it again.

**Comment by openanim** [comment]
Unfortunately in the state, I find that this functionality is awkward compared to its behavior in Harlequin. This should definitely be considered a bug. The very purpose of the path is unfortunately difficult to apply to the desired object, because of the offset. So bad considering its power and ease of use. :(.

**Comment by RodneyBaker** [comment]
Tagged as bug pending further discussion that demonstrates otherwise.
